sssd_test_framework/roles/ad.py

```diff
@@ -309,6 +344,29 @@ def test_example(client: Client, ad: AD):
 
         return ADSudoRule(self, name, basedn)
 
+    def attrs_to_hash(self, attrs: dict[str, Any]) -> str | None:
```

Comment:
Can you move it into misc/__init__.py and add unit test for it to tests/test_misc.py?

Reply:
Haha, you are right. Yes, it makes sense to keep it in AD because we use it only there. However, putting it to misc will allow you to unit test it and this time my mind leaned to this :-) It's your call, whatever you prefer.